packages/lib/src/application/App.ts

```diff
@@ -37,27 +37,27 @@ export default class App {
 
   constructor(appContract: ContractWrapper, txParams: any = {}) {
     this.appContract = appContract;
-    this.txParams = txParams;
+    this.txParams = { ...txParams };
   }
 
   public async getPackage(name): Promise<{ package: Package, version: string }> {
-    const [address, version] = await this.appContract.getPackage(name);
-    const thepackage = await Package.fetch(address, this.txParams);
+    const { ['0']: address, ['1']: version } = await this.appContract.methods.getPackage(name).call();
+    const thepackage = await Package.fetch(address, { ...this.txParams });
```

Comment:
Can we push further down the need to clone txParams? I'd try to do it only on the calls to web3, and not when passing them through our intermediate objects.